src/code.c

```diff
@@ -977,14 +977,57 @@ void CodeIfEnd (
     Stat                body;           /* body of a branch                */
     UInt                hase;           /* has else branch                 */
     UInt                i;              /* loop variable                   */
+    Expr                cond1 = 0;      /* first condition                 */
+    Expr                cond2 = 0;      /* second condition                */
 
-    /* peek at the last condition                                          */
+    /* peek at the last two conditions                                     */
     body = PopStat();
-    cond = PopExpr();
-    hase = (TNUM_EXPR(cond) == T_TRUE_EXPR);
-    PushExpr( cond );
+    cond1 = PopExpr();
+    hase = (TNUM_EXPR(cond1) == T_TRUE_EXPR);
+    if (nr == 2) {
+        Stat body2 = PopStat();
+        cond2 = PopExpr();
+        PushExpr(cond2);
+        PushStat(body2);
```

Comment:
You don't really need to pop/push `body2` (and the original code also did not need to push/pop `body`...